src/index.tsx

```diff
@@ -20,122 +25,136 @@ export default class SignatureCanvas extends Component {
     clearOnResize: PropTypes.bool
   }
 
-  static defaultProps = {
+  static defaultProps: Pick<ReactSignatureCanvasProps, 'clearOnResize'> = {
     clearOnResize: true
   }
 
-  _sigPad = null
+  // this is some hack-ish init and casting to avoid `| null` everywhere :/
+  /* eslint-disable @typescript-eslint/consistent-type-assertions */
+  _sigPad: SignaturePad = {} as SignaturePad
+  _canvas: HTMLCanvasElement = {} as HTMLCanvasElement
+  /* eslint-enable @typescript-eslint/consistent-type-assertions */
```

Comment:
This should best be avoided as it can lead to hard to track bugs in the future.

Reply:
It would require _a ton_ of refactoring to have `| null` work as there's accesses _everywhere_. 

I'm open to hearing other options, but I don't think that refactoring is 1) worth the time or 2) worth the added complexity/worse readability

Reply:
After giving it some thought: IMO this is a breaking change. This changes how `getSignaturePad` and `getCanvas` public API works, as now it returns empty object instead of null in some rare cases.

A workaround like the following could be used to avoid redundant code: 
```js
private getSignaturePadInstance: () => {
  if (this._sigPad == null) {
    throw new Error('some meaningful message');
  }

  return this._sigPad;
}
```

Reply:
Yea I recognize it could be _considered_ "breaking" in the _strict_ sense, but so can lots of things -- [ComVer](https://gitlab.com/staltz/comver#how-is-this-different-to-semver) exists for that reason.

Realistically that never happens and is definitely not behavior I'd encourage relying on or is really guaranteed by the library -- the docs do not mention that it might be `null`  and notably `@types/react-signature-canvas` does not handle the `null` case either.

I did consider adding a helper function, but that seemed far from ideal to me: to be explicit, adding `| null` requires changes in 19+ places in the code, or ~15%+ of the source. Some of those changes would be non-trivial and confusing, and they'd hamper readability IMO.

I didn't quite consider throwing an error though, and that probably makes more sense than just doing nothing, especially as these cases are likely to error anyway.
**_But_**, throwing an error now can be considered a breaking change as well, particularly so in the sense you're describing.
I'm also not really sure what it should say, that's a React edge case and would prefer to throw a React error but idk. Possibly "rSC is mounting/unmounting right now, React refs are null during this phase".
Also the `getSignaturePad` method already exists, there's no need for a separate method.

I don't know if there's a better solution; like I said, this seems far from ideal to me. Adding non-null assertion operators _everywhere_  would be the least "change" but also not ideal.

<hr>

The changes to `width`/`height` checks in `_resizeCanvas` actually do have potential to break things which is why I'm almost certainly going to revert them.

Reply:
Ok, throwing an error is definitely a lot cleaner. In several cases can also just assign an alias `const canvas = this.getCanvas()` to avoid repetition of the helper (but not in all places). Can push a change to this now, though still need to think about the ramifications of throwing an error.

Reply:
Throwing an error now. Seems ok. Only thing I wonder is if `getCanvas()` and `getSignaturePad()` shouldn't throw externally, and should just return `null` like a ref would?
I think a custom error is a lot easier to understand than, like, accidentally calling a method on `null`, but at the same time, changing the types to also return `null` would be more explicit and prevent that case too 🤔